File: net/src/Sdl.Web.GraphQL/Request/IGraphQLRequest.cs
﻿using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Sdl.Web.HttpClient.Auth;

namespace Sdl.Web.GraphQL.Request
{
    /// <summary>
    /// GraphQL Request
    /// </summary>
    public interface IGraphQLRequest
    {
        /// <summary>
        /// GraphQL Query
        /// </summary>
        string Query { get; set; }

        /// <summary>
        /// GraphQL Operation Name
        /// </summary>
        string OperationName { get; set; }

        /// <summary>
        /// GraphQL Variables
        /// </summary>
        IDictionary<string, object> Variables { get; set; }

        /// <summary>
        /// Authentication used for request
        /// </summary>
        [JsonIgnore]
        IAuthentication Authenticaton { get; set; }

        /// <summary>
        /// Add Variable
        /// </summary>
        /// <param name="name">Variable name</param>
        /// <param name="value">Variable value</param>
        /// <returns>The request</returns>
        IGraphQLRequest AddVariable(string name, object value);

        /// <summary>
        /// Serialization binder used when deserializing the request.
        /// </summary>
        [JsonIgnore]
        SerializationBinder Binder { get; set; }

        /// <summary>
        /// Convertor used when deserialzing the request
        /// </summary>
        [JsonIgnore]
        List<JsonConverter> Convertors { get; set; }
    }
}

File: java/GraphLClient/src/main/java/com/sdl/web/Request/IGraphQLRequest.java
package com.sdl.web.Request;

import com.sdl.web.Auth.IAuthentication;

import java.util.Dictionary;

public interface IGraphQLRequest {
    String Query = null;
    String OperationName = null;
    Dictionary<String, Object> Variables = null;
    IAuthentication Authenticaton = null;

    /* IGraphQLRequest AddVariable(String name, Object value);*/
    /*SerializationBinder Binder;*/
    /* List<JsonConverter> Convertors;*/

    void setQuery(String query);
    String getQuery();

    void setOperationName(String operationName);
    String getOperationName();

    void setVariables(Dictionary<String, Object> variables);
    Dictionary<String, Object> getVariables();

    void setAuthenticaton(IAuthentication authenticaton);
    IAuthentication getAuthenticaton();

}
